Update istio to 1.21.0 (#676)
kyma-project/istio: docs/release-notes/1.5.0.md

## New Features

- Update the Istio version to 1.21.0 [#676](https://github.com/kyma-project/istio/pull/676). Read [Istio 1.21.0 Release Announcement](https://istio.io/latest/news/releases/1.21.x/announcing-1.21/) and [Change Notes](https://istio.io/latest/news/releases/1.21.x/announcing-1.21/change-notes/) for more details.
- The version details for Istio installation are now only specified in the files located in the [`istiooperator`](https://github.com/kyma-project/istio/blob/main/internal/istiooperator) directory [#664](https://github.com/kyma-project/istio/pull/664).

> ## Breaking Changes in Istio upstream

Reviewer: ```suggestion
## Breaking Changes in Istio Upstream
```

> ## Breaking Changes in Istio Upstream

Reviewer: ```suggestion
## Relevant Changes in Istio Upstream
```

> - Istio no longer creates `IstioOperator/installed-state` CustomResource after installation. See the upstream [Istio Issue](https://github.com/istio/istio/issues/44814) and [Pull Request](https://github.com/istio/istio/pull/47860) for more information.

Reviewer: ```suggestion
- Istio no longer creates `IstioOperator/installed-state` CustomResource after installation. See the upstream [Istio issue](https://github.com/istio/istio/issues/44814) and the [PR](https://github.com/istio/istio/pull/47860) for more information.
```

> - Discontinue support for `Referer` headers containing user info or fragment components. The new sanitization behaviour is described [here](https://www.envoyproxy.io/docs/envoy/v1.26.7/configuration/http/http_conn_man/headers#config-http-conn-man-headers-referer).

Reviewer: ```suggestion
- Discontinue support for `Referer` headers containing user info or fragment components. The new sanitization behavior is described in the [Envoy's documentation](https://www.envoyproxy.io/docs/envoy/v1.26.7/configuration/http/http_conn_man/headers#config-http-conn-man-headers-referer).
```
